build.assets/grpcbox.mk

```diff
@@ -14,15 +14,18 @@ else
 GRPCBOX ?= $(GRPCBOX_BASE_NAME):$(BUILDBOX_VERSION)
 endif
 
+# Allow overriding the docker implementation to use for ex. podman.
+DOCKER := docker
```

Comment:
```suggestion
DOCKER ?= docker
```

Reply:
Works in both cases, but I can change it. TBH I was a bit supprised that `:=` allows overriding :thinking:

Reply:
Thanks to ChatGPT :see_no_evil: 

In makefiles, `:=`, `?=` and `=` are used to assign values to variables, but they do it in different ways.

1. `=` (Recursive Variable Assignment)

   `=` is used for recursive variable assignment. In this case, the value on the right-hand side of the equals sign is not evaluated immediately. Instead, it is stored as is and evaluated each time the variable is used.

    Here is an example:

    ```
    FOO = $(BAR)
    BAR = hello
    all:
        @echo $(FOO)
    ```

    When you run `make all`, it will print `hello`, not `$(BAR)`. This is because `$(FOO)` is evaluated each time it is used, and at the time of its use `$(BAR)` equals `hello`.

2. `:=` (Simple Variable Assignment)

   `:=` is used for simple variable assignment. The right-hand side is evaluated immediately and stored as the variable's value.

    Here is an example:

    ```
    FOO := $(BAR)
    BAR = hello
    all:
        @echo $(FOO)
    ```

    When you run `make all`, it will print an empty line, not `hello`. This is because `$(BAR)` was not defined at the time `FOO := $(BAR)` was evaluated.

3. `?=` (Conditional Variable Assignment)

   `?=` is used for conditional variable assignment. It assigns a value to a variable only if the variable is not defined yet.

    Here is an example:

    ```
    FOO ?= bar
    FOO ?= baz
    all:
        @echo $(FOO)
    ```

    When you run `make all`, it will print `bar`, not `baz`. This is because `FOO` was already defined when `FOO ?= baz` was encountered.
   
This is a quick overview. The GNU make manual contains more detailed information about these and other aspects of makefiles.


And


Variables assigned with `:=` in a Makefile can be overridden from the command line.

Let's say you have a Makefile like this:

```
FOO := bar
all:
	@echo $(FOO)
```

If you run `make all`, it will print `bar`. But if you run `make all FOO=baz`, it will print `baz`. This is because command-line arguments to `make` take precedence over assignments in the Makefile, regardless of the type of assignment used (`=`, `:=`, `?=`, etc.).

Here's an excerpt from the GNU `make` manual on this topic:

> Variables in make can come from the environment in which make is run. Every environment variable that make sees when it starts up is transformed into a make variable with the same name and value. However, an explicit assignment in the makefile, or with a command argument, overrides the environment. (If the `-e` flag is specified, then values from the environment override assignments in the makefile.)

This is why command-line arguments can override variables assigned with `:=` (or any other assignment operator) in the Makefile.